groups/bdl/bdlb/bdlb_pcgrandomgenerator.h

```diff
@@ -0,0 +1,307 @@
+// The PcgRandomGenerator constructor takes two 64-bit constants (the initial
+// state, and the rng sequence selector.  Instances of 'PcgRandomGenerator' with
+// different sequence selectors will never have random sequences that coincide.
+// In order to obtain nondeterministic output for each instance, on a Unix
+// system random bytes can be obtained from '/dev/urandom'.  The component 
+// bdlb_randomdevice provides such functionality.  Also illustrated below is a 
+// a backup mechanism for obtaining random bytes.
+//..
+//  uint64_t state;
+//  if (0 != RandomDevice::getRandomBytes((unsigned char *)&state,
+//                                                         sizeof(state))) {
+//          state = time(NULL) ^ (intptr_t)&bsl::printf;  // fallback state
+//  }
+//  uint64_t streamSelector;
+//  if (0 != RandomDevice::getRandomBytes((unsigned char *)&streamSelector,
+//                                              sizeof(streamSelector))) {
+//           streamSelector = time(NULL) ^
+//                   (intptr_t)&bsl::printf;  // fallback streamSelector
+//  }
+//  bdlb::PcgRandomGenerator  rng(state, streamSelector);
+//  bsl::uint32_t randomInt = rng.generate();
+//..
+
+#include <bsl_cstdint.h>
+
+namespace BloombergLP {
+namespace bdlb {
+
+                                 // ========================
+                                 // class PcgRandomGenerator
+                                 // ========================
+class PcgRandomGenerator {
+    // This class implements a random number generator (RNG) based on the PCG
+    // algorithm.  PCG stands for "permuted congruential generator." The state
+    // is 64 bits.  It uses a so-called stream selector, also 64 bits. For
+    // details, please refer below to the constructor.  For details of the
+    // algorithm, see http://www.pcg-random.org.
+
+  private:
+    // DATA
+    bsl::uint64_t d_state;           // the RNG state
+
+    bsl::uint64_t d_streamSelector;  // selected sequence (stream)
+
+    // FRIENDS
+    friend bool operator==(const PcgRandomGenerator& lhs,
+                           const PcgRandomGenerator& rhs);
+
+  public:
+    // CREATORS
+    PcgRandomGenerator(bsl::uint64_t initState      = 0,
+                       bsl::uint64_t streamSelector = 1);
+        // Create a 'PcgRandomGenerator' object and seed it with the optionally
+        // specified 'initState' and 'streamSelector.'  If 'initState' is not
+        // specified, 0 is used as the initial state.  If 'streamSelector' is
+        // not specified, 1 is used as the stream selector. 'initState' is the
+        // starting state for the RNG.  Any 64-bit value may be passed.
+        // 'streamSelector' selects the output sequence for the RNG.  Any
+        // 64-bit value may be passed, although only the low 63 bits are
+        // significant.  There are 2^63 different RNGs available, and
+        // 'streamSelector' selects from among them.  Invoking different
```

Comment:
Too much information here, which is at the same time unclear. Most of it covered in component doc, so let's keep this simple.  

```suggestion
        // not specified, 1 is used as the stream selector.  Only the 
        // low 63 bits of 'streamSelector' are
        // significant.  Note that invoking different
```